File: data/EXP01/golang_no_system_var/2467/2467.output_1.go
func mostProfitablePath(edges [][]int, bob int, amount []int) int {
    n := len(amount)
    
    adjList := make([][]int, n)
    for _, edge := range edges {
        u, v := edge[0], edge[1]
        adjList[u] = append(adjList[u], v)
        adjList[v] = append(adjList[v], u)
    }
    
    visited := make([]bool, n)
    res := make([]int, n)
    
    dfs(0, adjList, visited, amount, res)
    
    return res[bob]
}

func dfs(node int, adjList [][]int, visited []bool, amount []int, res []int) {
    visited[node] = true
    result := 0
    
    for _, neighbor := range adjList[node] {
        if !visited[neighbor] {
            dfs(neighbor, adjList, visited, amount, res)
            result = max(result, res[neighbor])
        }
    }
    
    res[node] = result + amount[node]
}

func max(a, b int) int {
    if a > b {
        return a
    }
    return b
}

File: data/EXP01/python_2/2467/2467.output_1.py
        class Solution:
            def mostProfitablePath(self, edges: List[List[int]], bob: int, amount: List[int]) -> int:
                graph = [[] for _ in range(len(amount))]
                for edge in edges:
                    graph[edge[0]].append(edge[1])
                    graph[edge[1]].append(edge[0])
                
                def dfs(node, parent):
                    max_profit = amount[node]
                    for neighbor in graph[node]:
                        if neighbor != parent:
                            profit = dfs(neighbor, node)
                            max_profit += max(0, profit)
                    return max_profit
                
                return dfs(bob, -1)
        